### PR TITLE
Jitera/custom props

### PR DESCRIPTION
**Pull request by viettran1010**
- added Extension for custom props on block level
- used Mark for custom props on content level 
- so we will need to do conversion at studio level, ie. putting all the extra fields from BE to customProps (block level) and customContentProps (content level), and when sending data to BE, we do extract data again from blocknote field to match the payload of BE

In App.tsx I created mock data with custom fields and set to editor, when editor blocks changed we can see in the console.log of the editor.topLevelBlocks still has all these fields

https://github.com/Jitera-Product/BlockNote/assets/48788462/304f13fc-fa38-4bd5-8ee1-b738e59e91ae

### CODE REVIEW

File: examples/editor/src/App.tsx [hunk 1]
```diff
@@ -17,6 +19,73 @@ function App() {
...
+          parent_id: "29",
+        },
+      },
+      children: [],
+    },
+  ];
+
+  useEffect(() => {
+    editor.replaceBlocks(editor.topLevelBlocks, convertedBlocks);
+    console.log(editor.topLevelBlocks);
```
Comment (zrg-team) [inline examples/editor/src/App.tsx:87]: Please remove log

File: examples/editor/src/main.tsx [hunk 2]
```diff
@@ -6,8 +6,4 @@ import "./index.css";
 window.React = React;
 
 const root = createRoot(document.getElementById("root")!);
-root.render(
-  <React.StrictMode>
```
Comment (zrg-team) [inline examples/editor/src/main.tsx:10]: Could we keep trict mode. Dont worry about render 2 times

**Review by zrg-team — APPROVED** [review]

**Review by jefrydco — APPROVED** [review]
Good work!

File: examples/editor/src/App.tsx (hunk 1 above)
Comment (viettran1010) [reply examples/editor/src/App.tsx:87]: we are not using @blocknote/react so this is for debug purpose

File: examples/editor/src/main.tsx (hunk 2 above)
Comment (viettran1010) [reply examples/editor/src/main.tsx:10]: same, we are not using @blocknote/react so this is for the logs to not flood

**Pull request merged**